streamingccl: fix histogram metrics

msbutler opened the pull request:
See individual commits.

Epic: none

Release note: none

msbutler committed:
streamingccl: update LDR histogram metrics

This patch does a few things:
-  updates the histogram metrics to use the preferred histogram mode
`metric.HistogramModePrometheus.`
- bumps the max commit latency histogram value to an hour
- aggregates the commit latency graphs over the max latency over the nodes

Epic: none

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/125698)
<!-- Reviewable:end -->

msbutler committed:
streamingccl: log replicatied time as go time, not AOST

Epic: none

Release note: none

msbutler committed:
streamingccl: ensure logical ingestion buffer inits with max timestamp

Without this patch, the logical_replication.commit_latency metric will look
wonky.

Epic: none

Release: none

msbutler commented:
TFTR!

bors r=dt

stevendanna commented:
We might want to do something similar in int PCR when we have a free moment.
